test/jdk/java/net/URLClassLoader/JarLoaderCloseTest.java

```diff
@@ -0,0 +1,158 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+import java.util.stream.Stream;
+
+import jdk.test.lib.util.JarUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+/*
+ * @test
+ * @bug 8338445
+ * @summary verify that the jdk.internal.loader.URLClassPath closes the JarFile
+ *          instances that it no longer uses for loading
+ * @library /test/lib
+ * @build jdk.test.lib.util.JarUtils
+ * @comment This test expects MalformedURLException for some specific URLs.
+ *          We use othervm to prevent interference from other tests which
+ *          might have installed custom URLStreamHandler(s)
+ * @run junit/othervm JarLoaderCloseTest
+ */
+public class JarLoaderCloseTest {
+
+    private static final String RESOURCE_NAME = "foo-bar.txt";
+    private static final String RESOURCE_CONTENT = "Hello world";
+    private static final Path TEST_SCRATCH_DIR = Path.of(".");
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        // create a file which will be added to the jar that gets tested
+        Files.writeString(TEST_SCRATCH_DIR.resolve(RESOURCE_NAME), RESOURCE_CONTENT);
+    }
+
+    static Stream<Arguments> malformedClassPaths() {
+        return Stream.of(
+                Arguments.of("C:\\foo\\bar\\hello/world.jar lib2.jar"),
+                Arguments.of("C:/hello/world/foo.jar"),
+                Arguments.of("lib4.jar C:\\bar\\foo\\world/hello.jar")
+        );
+    }
+
+    /*
+     * Creates a jar with a manifest which has a Class-Path entry value with malformed URLs.
```

Comment:
I think you mean a "a JAR file" rather than "jar" here. Same nit in  testParsableClassPathEntry.

Reply:
Fixed.